### PR TITLE
LPS-146419 Fix view URL generation for non-default language at info panel in content dashboard

### PR DESCRIPTION
**Pull request by YurenaC**
Motivation
=======
A [bug](https://issues.liferay.com/browse/LPS-146419) detected by our QA team highlighted a problem with the visualization of web contents from the information panel in the content dashboard when those web contents have translations. 
All the visualizations lead to the default web content in English

Proposed Solution
========
There are 3 different modes for friendly url styles. The mode number 3 is defined by default within the portal but it wasn't even been consider in the url generation.
It has been added and the issue solved.

Steps to Verify:
----------------
**Requirement:**

A default Display Page Template created, containing a mapped 'Heading' fragment
One Web Content with a translation (e.g. Spanish)

**Step-by-Step:**

Open product menu > Applications > Content > Content Dashboard
At Content, hover over the asset > Click on 'Info' icon
At sidebar, expand 'Details' > Click on 'View' icon of the respective translation

Change summary:
---------------
*  LPS-146419 Also considering the local friendly url style number 3 when creating view URL for different languages than the default

### CODE REVIEW

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Review by cgoncas — CHANGES_REQUESTED** [review]
Do you mind create a test for this change, thanks @YurenaC

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
Hi @cgoncas ,
Test added, at the end is an integration test, unit ones were so difficult to mock :(

**Comment by YurenaC** [comment]
ci:test:tango

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:stable

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Hey @YurenaC !!!

Do you mind delete the testCompile, as it is the error you are seeing in the logs for this PR:

junit.framework.AssertionFailedError: Please delete the unused testCompile dependencies in modules/apps/layout/layout-type-controller/layout-type-controller-asset-display/build.gradle

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
Branch updated with master and tests rerun

**Review by cgoncas — CHANGES_REQUESTED** [review]

File: modules/apps/layout/layout-type-controller/layout-type-controller-asset-display/src/main/java/com/liferay/layout/type/controller/asset/display/internal/portlet/AssetDisplayPageFriendlyURLProviderImpl.java [hunk 1]
```diff
@@ -167,7 +167,9 @@ private void _setThemeDisplayI18n(
 		if ((languageIds.contains(locale.toString()) &&
 			 (PropsValues.LOCALE_PREPEND_FRIENDLY_URL_STYLE == 1) &&
 			 !locale.equals(LocaleUtil.getDefault())) ||
-			(PropsValues.LOCALE_PREPEND_FRIENDLY_URL_STYLE == 2)) {
+			(PropsValues.LOCALE_PREPEND_FRIENDLY_URL_STYLE == 2) ||
```
[the remark this answers is not in the document]
Comment (YurenaC) [reply modules/apps/layout/layout-type-controller/layout-type-controller-asset-display/src/main/java/com/liferay/layout/type/controller/asset/display/internal/portlet/AssetDisplayPageFriendlyURLProviderImpl.java:170]: Hi @cgoncas, 
I also checked the https://issues.liferay.com/browse/LPS-150919

Applying the same change I made in `AssetDisplayPageFriendlyURLProviderImpl` to `PortalImpl` the issue is solved
Urls

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
@cgoncas, the code has been updated based on our conversation, please review!

**Pull request closed without merging**